### ISSUE TITLE
Quick thoughts/feedback

### ISSUE THREAD

**Issue by jbenet** [hsanjuan/go-libp2p-gorpc#1]
This module is really cool :+1:.

Wonder if we should rename it as `go-libp2p-gorpc` (because `js-libp2p-rpc` would probably NOT mean the same thing, i.e. it probably would not use go types and gob, (even though i personally think there's ample good reason to use go types and gob as a cross language RPC framework, instead of something like gRPC).

@whyrusleeping and I discussed making `libp2p-grpc` in the past. wonder if now's a good time do that. It may cause a bunch of libp2p adoption, because people want grpc in a bunch of other transports (quic, webrtc, websockets, NOT http). We should discuss this in detail in the coming weeks (pros/cons, etc). @hsanjuan have you had experience with gRPC much? i am personally not very sold on gRPC because the views/abstractions of gRPC/protobuf conflict with my views/abstractions (eg i much prefer go/rpc, that fits my views/abstractions better), but i do like and recognize the huge pragmatic value they provide.

**Comment by jbenet** [comment]
or maybe `go-libp2p-gobrpc`. (i prefer `-gorpc`, easier to say, but it may be clearer to say `-gobrpc`)

**Comment by hsanjuan** [comment]
I don't have experience with gRPC. I can allocate some time.

  * `-gob` suggests this uses Gob? But it uses https://godoc.org/github.com/ugorji/go/codec + msgpack. 

  * `net/rpc` offers a codec abstraction. I didn't copycat that part because we have a codec abstraction: multicodecs. But I hit some bumps with multicodecs and libp2p-raft so I ended up using what worked for me quickly (here too). Last time I looked, I did not find a way for multicodecs to take Reader/Writer() like Gob does. The  example does it: https://github.com/multiformats/go-multicodec . But it seems there is no NewEncoder()/NewDecoder() methods (https://godoc.org/github.com/multiformats/go-multicodec#Multicodec) and I can't run it. I need to allocate time to track this down and report/fix it in the appropiate places though.
